packages/slate-base64-serializer/src/index.js

```diff
@@ -1,5 +1,49 @@
 
 import { State } from 'slate'
+import { Buffer } from 'buffer'
+
+function newBuffer(data, encoding, len) {
+  return new Buffer(data, encoding, len)
+}
+
+if (!Buffer.from) {
+  Buffer.from = newBuffer
+}
+
+/**
+ * encode string `str` to base64
+ *
+ * @param  {String} str
+ * @return {String}
+ */
+
+function btoa(str) {
+  let buffer
+  if (window && window.btoa) {
```

Comment:
will throw on server since window is undefined. better ignore window and use `typeof btoa !== 'undefined'`

Reply:
it won't throw an error since it checks if window is there or not as well. this way window.btoa(client side btoa) will still work but it it doesn't exist then use Buffer conversion

Reply:
If you check for window reference on nodejs, you will run into a ReferenceError since window is not defined. Just try for yourself and run this through node:
```js
if (!window) { console.log('hi') } // => will throw
if (typeof window === 'undefined') { console.log('hi') } // => will work
```

Reply:
good pickup, didn't know that window check throw an error on server side